### DISCUSSION TITLE
Github Action - Prompting inputs

### DISCUSSION THREAD

**Discussion by pierreminiggio** [community/community#19260] [Actions]
Hello ! I developed an action that requires 2FA Auth (EDIT : Not really 2FA Auth actually, more like a code that is sent to an email).
I want to be able to prompt an input during the execution of my action (because right now my action simply dies whenever I'm required to provide the code I receive, cauz' I can't input it).
And from what I've researched, it doesn't seem to be an available feature in the current state of Github Actions.
Could this be implemented ?

I guess I could work around it by sending a request to a server I own and channel that into a telegram bot or something so that I can input the code there whenever I need to, but that's really an overkill solution just to prompt an input.

Can I get a better solution ?

**Comment by grafail** [comment]
You can probably use something like "oathtool", put the 2FA private key in a github secret and generate the 2fa token inside the action.

**Reply by pierreminiggio** [reply]
Thanks for the input (pun intended :D) !
Sadly I used the wrong word in my original post, it's not really 2FA, but more like a verification code that is sent via email, so I can't change the way the code is generated :'(
To be even more precise, it's Instagram's email confirmation code on login.
So yeah, I don't think the solution you suggested works for my usecase :'(

**Reply by marcc-cofomo** [reply]
I have a similar need: I want a Terraform workflow to stop after the `plan` phase and wait for an admin to confirm the plan before `apply` is ran. A workaround would be to have the plan stashed as an artifact and have a secondary, manually triggered workflow consume it as an input, but it would be much more convenient and user-friendly to have it all happen in one place.

**Reply by tasktop-teho** [reply]
> I have a similar need: I want a Terraform workflow to stop after the `plan` phase and wait for an admin to confirm the plan before `apply` is ran. A workaround would be to have the plan stashed as an artifact and have a secondary, manually triggered workflow consume it as an input, but it would be much more convenient and user-friendly to have it all happen in one place.

We have been trying https://github.com/marketplace/actions/manual-workflow-approval to approve Terraform plans. It uses Github issues as the UI to approve or deny.

**Comment by touficbatache** [comment]
Salut @pierreminiggio, were you able to find a solution for this? Running into the same problem 😕

**Reply by pierreminiggio** [reply]
The workaround I used was to set up a REST API that connects to a Telegram bot.
When my API is called, it sends a text to Telegram channel I created, then it listens to messages sent into that channel, the next message will be me inputing the code using my Telegram app, and then the API returns what I just typed in said message.

Of course I had to setup long timeout for both the API call, and the API response time while it waits for me to send the code on Telegram.

I did that in PHP, it looks like this if you're curious :
https://github.com/pierreminiggio/instagram-code-beggar-api/blob/main/src/App.php

Technically, you could do that within your Github Action's code base directly rather than setting up an external API for this, given it's just HTTP requests.
